registry/handlers/blob.go

```diff
@@ -76,16 +76,16 @@ func (bh *blobHandler) DeleteBlob(w http.ResponseWriter, r *http.Request) {
 	err := blobs.Delete(bh, bh.Digest)
 	if err != nil {
 		switch err {
-		case distribution.ErrBlobUnknown:
-			w.WriteHeader(http.StatusNotFound)
-			bh.Errors = append(bh.Errors, v2.ErrorCodeBlobUnknown)
 		case distribution.ErrUnsupported:
-			w.WriteHeader(http.StatusMethodNotAllowed)
 			bh.Errors = append(bh.Errors, v2.ErrorCodeUnsupported)
+			return
+		case distribution.ErrBlobUnknown:
+			bh.Errors = append(bh.Errors, v2.ErrorCodeBlobUnknown)
+			return
 		default:
 			bh.Errors = append(bh.Errors, errcode.ErrorCodeUnknown)
```

Comment:
The original `err.String()` should be included in the `Detail` or be logged to provide some way of debugging failed deletes. This line can actually be this:

```
append(bh.Errors, err)
```

`ErrorCodeUnknown` will be automatically populated.